### ISSUE TITLE
Suggestions about the node search box.

### ISSUE THREAD

**Issue by wibur0620** [comfyanonymous/ComfyUI#2716]
https://github.com/comfyanonymous/ComfyUI/assets/131615326/c0026c40-580f-47f8-9b8e-53a603c3f3c2

Based on the situation shown in the video, when the mouse leaves the scope of the node search box, it disappears automatically. However, for Chinese users, there is a serious problem, and this might caused node search box to disappear when Chinese users try to input Chinese using Chinese input method,due to the checkbox pop-up from the Chinese input method, which puts the mouse cursor above the checkbox of the Chinese input method. Therefore, I suggest that the node search box should be cancelled by clicking when the mouse is outside the node search box range. Instead of automatically canceling the node search box when the mouse leaves the node search box.

**Comment by NeedsMoar** [comment]
Yeah, it's incredibly easy to do by accident typing in English.  I have my mouse sensitivity up pretty high because of monitor resolution and sometimes I'll just let go of the mouse, type something, then accidentally move the cursor 10 pixels when I grab it again and the search is gone.  Or the cat will decide to hop up and bump it and move things mid-typing.

**Comment by zaZaza7979** [comment]
Isn't there a way to turn this feature on/off in the options? One more problem is that pressing enter after finding the desired node in the search box doesn't create the node, so I always have to alternate between mouse and keyboard.

**Comment by NeedsMoar** [comment]
As an example of how this is done by high end software, Houdini has a fairly similar (except better organized, and it fits in 1/4 of my screen easily) node selection menu containing something like as many nodes as a comfy install would have if you somehow installed every third party module and it still worked.  Although it limits the displayed nodes to those applicable to the current scope that count still tops what I have installed in comfy in most of the scopes because Houdini is insane.   Anyway you can access it by right clicking or hitting tab in the node window...  It's more the equivalent of Comfy's main node list, except that it has the search built in and you can just start typing and start narrowing down results immediately from that menu rather than a different one...  it's sort of like the Windows Key + Start typing to search the start menu (and everything else on earth if you didn't disable the annoying internet search) on Windows.   

Moving the mouse away from it does nothing, you have to click somewhere off of it to clear it.  It actually becomes modal within Houdini and no other controls will highlight or do anything from mouseover, only from clicking away.  This is a good thing because the entire workflow is node based so when you have that menu open you want it to stay open until you find what you needed, change your mind, or realize that you needed to create the node within some other context.  Comfy doesn't have the contexts so that's one less reason to not just integrate search into the main menu and keep it locked open until you click somewhere else.
